{
  "uid" : "ae34a9d8ff65ac31",
  "name" : "test_get_allowance[Binance Smart Chain (BNB)]",
  "fullName" : "tests.api_tests.tests_de_bridge#test_get_allowance",
  "historyId" : "d8ad6a5332a8d6736c1266d6f5e8abbe",
  "time" : {
    "start" : 1685615508956,
    "stop" : 1685615510134,
    "duration" : 1178
  },
  "status" : "broken",
  "statusMessage" : "jsonschema.exceptions.ValidationError: '0' is not of type 'array'\n\nFailed validating 'type' in schema['properties']['data']:\n    {'items': {'properties': {'address': {'type': 'string'},\n                              'decimals': {'type': 'integer'},\n                              'logoURI': {'type': 'string'},\n                              'name': {'type': 'string'},\n                              'symbol': {'type': 'string'}},\n               'required': ['symbol',\n                            'name',\n                            'decimals',\n                            'address',\n                            'logoURI'],\n               'type': 'object'},\n     'type': 'array'}\n\nOn instance['data']:\n    '0'",
  "statusTrace" : "network = 'Binance Smart Chain (BNB)'\n\n    @pytest.mark.parametrize(\"network\", networks)  # todo разобрать ошибку\n    def test_get_allowance(network):  # \"error\":\"No available RPC's for ETH\"\n        net = de_bridge_keys_data()[network].get('net')\n        not_native_token = de_bridge_keys_data()[network].get('notNativeToken')\n        owner = de_bridge_keys_data()[network].get('owner')\n>       DeBridge().get_allowance(net=net, not_native_token=not_native_token, owner=owner)\n\ntests/api_tests/tests_de_bridge.py:49: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nserviceCalls/debrdige_service.py:82: in get_allowance\n    validate(json.loads(response.text),\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ninstance = {'data': '0', 'error': '', 'ok': True}\nschema = {'$schema': 'http://json-schema.org/draft-07/schema#', 'properties': {'data': {'items': {'properties': {'address': {'t...rray'}, 'error': {'type': 'string'}, 'ok': {'type': 'boolean'}}, 'required': ['ok', 'data', 'error'], 'type': 'object'}\ncls = <class 'jsonschema.validators.Draft7Validator'>, args = (), kwargs = {}\nvalidator = Draft7Validator(schema={'$schema': 'http://json-...ft-07/schema#', 'properties': {'data': {'items': {'properties': {'a...e': 'string'}, 'ok': {'type': 'boolean'}}, 'required': ['ok', 'data', 'error'], 'type': 'object'}, format_checker=None)\nerror = <ValidationError: \"'0' is not of type 'array'\">\n\n    def validate(instance, schema, cls=None, *args, **kwargs):\n        \"\"\"\n        Validate an instance under the given schema.\n    \n            >>> validate([2, 3, 4], {\"maxItems\": 2})\n            Traceback (most recent call last):\n                ...\n            ValidationError: [2, 3, 4] is too long\n    \n        :func:`~jsonschema.validators.validate` will first verify that the\n        provided schema is itself valid, since not doing so can lead to less\n        obvious error messages and fail in less obvious or consistent ways.\n    \n        If you know you have a valid schema already, especially\n        if you intend to validate multiple instances with\n        the same schema, you likely would prefer using the\n        `jsonschema.protocols.Validator.validate` method directly on a\n        specific validator (e.g. ``Draft20212Validator.validate``).\n    \n    \n        Arguments:\n    \n            instance:\n    \n                The instance to validate\n    \n            schema:\n    \n                The schema to validate with\n    \n            cls (jsonschema.protocols.Validator):\n    \n                The class that will be used to validate the instance.\n    \n        If the ``cls`` argument is not provided, two things will happen\n        in accordance with the specification. First, if the schema has a\n        :kw:`$schema` keyword containing a known meta-schema [#]_ then the\n        proper validator will be used. The specification recommends that\n        all schemas contain :kw:`$schema` properties for this reason. If no\n        :kw:`$schema` property is found, the default validator class is the\n        latest released draft.\n    \n        Any other provided positional and keyword arguments will be passed\n        on when instantiating the ``cls``.\n    \n        Raises:\n    \n            `jsonschema.exceptions.ValidationError`:\n    \n                if the instance is invalid\n    \n            `jsonschema.exceptions.SchemaError`:\n    \n                if the schema itself is invalid\n    \n        .. rubric:: Footnotes\n        .. [#] known by a validator registered with\n            `jsonschema.validators.validates`\n        \"\"\"\n        if cls is None:\n            cls = validator_for(schema)\n    \n        cls.check_schema(schema)\n        validator = cls(schema, *args, **kwargs)\n        error = exceptions.best_match(validator.iter_errors(instance))\n        if error is not None:\n>           raise error\nE           jsonschema.exceptions.ValidationError: '0' is not of type 'array'\nE           \nE           Failed validating 'type' in schema['properties']['data']:\nE               {'items': {'properties': {'address': {'type': 'string'},\nE                                         'decimals': {'type': 'integer'},\nE                                         'logoURI': {'type': 'string'},\nE                                         'name': {'type': 'string'},\nE                                         'symbol': {'type': 'string'}},\nE                          'required': ['symbol',\nE                                       'name',\nE                                       'decimals',\nE                                       'address',\nE                                       'logoURI'],\nE                          'type': 'object'},\nE                'type': 'array'}\nE           \nE           On instance['data']:\nE               '0'\n\nvenv/lib/python3.9/site-packages/jsonschema/validators.py:1121: ValidationError",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ ],
  "testStage" : {
    "status" : "broken",
    "statusMessage" : "jsonschema.exceptions.ValidationError: '0' is not of type 'array'\n\nFailed validating 'type' in schema['properties']['data']:\n    {'items': {'properties': {'address': {'type': 'string'},\n                              'decimals': {'type': 'integer'},\n                              'logoURI': {'type': 'string'},\n                              'name': {'type': 'string'},\n                              'symbol': {'type': 'string'}},\n               'required': ['symbol',\n                            'name',\n                            'decimals',\n                            'address',\n                            'logoURI'],\n               'type': 'object'},\n     'type': 'array'}\n\nOn instance['data']:\n    '0'",
    "statusTrace" : "network = 'Binance Smart Chain (BNB)'\n\n    @pytest.mark.parametrize(\"network\", networks)  # todo разобрать ошибку\n    def test_get_allowance(network):  # \"error\":\"No available RPC's for ETH\"\n        net = de_bridge_keys_data()[network].get('net')\n        not_native_token = de_bridge_keys_data()[network].get('notNativeToken')\n        owner = de_bridge_keys_data()[network].get('owner')\n>       DeBridge().get_allowance(net=net, not_native_token=not_native_token, owner=owner)\n\ntests/api_tests/tests_de_bridge.py:49: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nserviceCalls/debrdige_service.py:82: in get_allowance\n    validate(json.loads(response.text),\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ninstance = {'data': '0', 'error': '', 'ok': True}\nschema = {'$schema': 'http://json-schema.org/draft-07/schema#', 'properties': {'data': {'items': {'properties': {'address': {'t...rray'}, 'error': {'type': 'string'}, 'ok': {'type': 'boolean'}}, 'required': ['ok', 'data', 'error'], 'type': 'object'}\ncls = <class 'jsonschema.validators.Draft7Validator'>, args = (), kwargs = {}\nvalidator = Draft7Validator(schema={'$schema': 'http://json-...ft-07/schema#', 'properties': {'data': {'items': {'properties': {'a...e': 'string'}, 'ok': {'type': 'boolean'}}, 'required': ['ok', 'data', 'error'], 'type': 'object'}, format_checker=None)\nerror = <ValidationError: \"'0' is not of type 'array'\">\n\n    def validate(instance, schema, cls=None, *args, **kwargs):\n        \"\"\"\n        Validate an instance under the given schema.\n    \n            >>> validate([2, 3, 4], {\"maxItems\": 2})\n            Traceback (most recent call last):\n                ...\n            ValidationError: [2, 3, 4] is too long\n    \n        :func:`~jsonschema.validators.validate` will first verify that the\n        provided schema is itself valid, since not doing so can lead to less\n        obvious error messages and fail in less obvious or consistent ways.\n    \n        If you know you have a valid schema already, especially\n        if you intend to validate multiple instances with\n        the same schema, you likely would prefer using the\n        `jsonschema.protocols.Validator.validate` method directly on a\n        specific validator (e.g. ``Draft20212Validator.validate``).\n    \n    \n        Arguments:\n    \n            instance:\n    \n                The instance to validate\n    \n            schema:\n    \n                The schema to validate with\n    \n            cls (jsonschema.protocols.Validator):\n    \n                The class that will be used to validate the instance.\n    \n        If the ``cls`` argument is not provided, two things will happen\n        in accordance with the specification. First, if the schema has a\n        :kw:`$schema` keyword containing a known meta-schema [#]_ then the\n        proper validator will be used. The specification recommends that\n        all schemas contain :kw:`$schema` properties for this reason. If no\n        :kw:`$schema` property is found, the default validator class is the\n        latest released draft.\n    \n        Any other provided positional and keyword arguments will be passed\n        on when instantiating the ``cls``.\n    \n        Raises:\n    \n            `jsonschema.exceptions.ValidationError`:\n    \n                if the instance is invalid\n    \n            `jsonschema.exceptions.SchemaError`:\n    \n                if the schema itself is invalid\n    \n        .. rubric:: Footnotes\n        .. [#] known by a validator registered with\n            `jsonschema.validators.validates`\n        \"\"\"\n        if cls is None:\n            cls = validator_for(schema)\n    \n        cls.check_schema(schema)\n        validator = cls(schema, *args, **kwargs)\n        error = exceptions.best_match(validator.iter_errors(instance))\n        if error is not None:\n>           raise error\nE           jsonschema.exceptions.ValidationError: '0' is not of type 'array'\nE           \nE           Failed validating 'type' in schema['properties']['data']:\nE               {'items': {'properties': {'address': {'type': 'string'},\nE                                         'decimals': {'type': 'integer'},\nE                                         'logoURI': {'type': 'string'},\nE                                         'name': {'type': 'string'},\nE                                         'symbol': {'type': 'string'}},\nE                          'required': ['symbol',\nE                                       'name',\nE                                       'decimals',\nE                                       'address',\nE                                       'logoURI'],\nE                          'type': 'object'},\nE                'type': 'array'}\nE           \nE           On instance['data']:\nE               '0'\n\nvenv/lib/python3.9/site-packages/jsonschema/validators.py:1121: ValidationError",
    "steps" : [ ],
    "attachments" : [ {
      "uid" : "ac3dab7ba189027",
      "name" : "log",
      "source" : "ac3dab7ba189027.txt",
      "type" : "text/plain",
      "size" : 300
    } ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 1,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ ],
  "labels" : [ {
    "name" : "parentSuite",
    "value" : "tests.api_tests"
  }, {
    "name" : "suite",
    "value" : "tests_de_bridge"
  }, {
    "name" : "host",
    "value" : "MacBook-Air-Anna.local"
  }, {
    "name" : "thread",
    "value" : "18188-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.api_tests.tests_de_bridge"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ {
    "name" : "network",
    "value" : "'Binance Smart Chain (BNB)'"
  } ],
  "links" : [ ],
  "hidden" : true,
  "retry" : true,
  "extra" : {
    "categories" : [ ],
    "tags" : [ ]
  },
  "source" : "ae34a9d8ff65ac31.json",
  "parameterValues" : [ "'Binance Smart Chain (BNB)'" ]
}